criu/page-xfer.c

```diff
@@ -758,24 +763,25 @@ static long fill_userbuf(int pid, struct page_pipe_buf *ppb,
 	unsigned long partial_read_bytes = 0;
 
 	while (start < ppb->nr_segs) {
-
 		bytes_read = process_vm_readv(pid, bufvec, 1, &riov[start],
 						   ppb->nr_segs - start, 0);
-
 		if (bytes_read == -1) {
-			/* Handling Case 1*/
-			if (riov[start].iov_len == PAGE_SIZE) {
+			/* Handling Case 1 */
+			if (errno == EFAULT && riov[start].iov_len == PAGE_SIZE) {
 				cnt_sub(CNT_PAGES_WRITTEN, 1);
 				start += 1;
 				continue;
 			} else if (errno == ESRCH) {
-				pr_debug("Target process PID:%d not found\n", pid);
-				return ESRCH;
+				pr_err("Target process PID:%d not found\n", pid);
+				return -ESRCH;
+			} else {
+				pr_perror("process_vm_readv");
+				BUG();
+				return -errno;
```

Comment:
without the else statement the following test is passing
`./test/zdtm.py run --ignore-taint -t zdtm/transition/maps007 --pre 2 --pre-dump-mode=read -f h`

```diff --git a/criu/page-xfer.c b/criu/page-xfer.c
index 63cf4f783..d40d5a057 100644
--- a/criu/page-xfer.c
+++ b/criu/page-xfer.c
@@ -774,10 +774,6 @@ static ssize_t fill_userbuf(int pid, struct page_pipe_buf *ppb,
                        } else if (errno == ESRCH) {
                                pr_err("Target process PID:%d not found\n", pid);
                                return -ESRCH;
-                       } else {
-                               pr_perror("process_vm_readv");
-                               BUG();
-                               return -errno;
                        }
                }
```
running this test >100, it always passes on my machine. Could you provide more information about when `page-xfer: Missing 7f276c079000 in parent pagemap` may occur?